src/github.rs

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use anyhow::{bail, Context};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct Repo {
+    pub full_name: String,
+}
+
+pub fn has_starred(github_id: &str, repo_full_name: &str) -> anyhow::Result<bool> {
+    let response = ureq::get(&format!("https://api.github.com/user/{github_id}/starred")).call()?;
```

Comment:
I'm fine with this as a temporary solution, but this will eventually hit the rate limit and potentially can be pretty slow operation.
We should use https://docs.github.com/en/rest/activity/starring#check-if-a-repository-is-starred-by-the-authenticated-user instead with the token we should already have in the request